core/dbt/cli/resolvers.py

```diff
@@ -10,7 +10,8 @@ def default_project_dir() -> Path:
 
 
 def default_profiles_dir() -> Path:
-    return Path.cwd() if (Path.cwd() / "profiles.yml").exists() else Path.home() / ".dbt"
+    paths = [Path.cwd(), Path.home() / ".dbt"]
+    return next((x for x in paths if (x / "profiles.yml").exists()), Path.cwd())
```

Comment:
It looks to me like the **true** problem is that executing the `runner.py` script [here](https://github.com/dbt-labs/dbt-core/issues/11286#issuecomment-2757855349) is behaving differently than `dbt ls --profiles-dir dbt_dir` for some reason.

We should get to the root cause of that discrepancy (which might trace back to [here](https://github.com/dbt-labs/dbt-core/blob/300aa09fc562fef6dade08cc5cd7297d636efa3f/core/dbt/cli/main.py#L53)). Or maybe [here](https://github.com/dbt-labs/dbt-core/blob/300aa09fc562fef6dade08cc5cd7297d636efa3f/core/dbt/cli/flags.py#L253-L254).

This solution looks like it would just mask the underlying discrepancy, so I do **not** think we should adopt it until we fully understand the root cause.